health-survey/src/main/java/org/lukawska/trainsmart/healthsurvey/application/service/HealthSurveyService.java

```diff
@@ -0,0 +1,114 @@
+import org.lukawska.trainsmart.healthsurvey.domain.entites.HealthSurvey;
+import org.lukawska.trainsmart.healthsurvey.domain.repositories.HealthSurveyRepository;
+import org.lukawska.trainsmart.shared_persistence.application.service.UserService;
+import org.lukawska.trainsmart.shared_persistence.domain.entities.User;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+import static org.lukawska.trainsmart.healthsurvey.application.mapper.HealthSurveyMapper.mapToEntity;
+import static org.lukawska.trainsmart.healthsurvey.application.mapper.HealthSurveyMapper.mapToHealthSurveyResponse;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class HealthSurveyService {
+
+    private final HealthSurveyRepository healthSurveyRepository;
+
+    private final UserService userService;
+
+    @Transactional
+    public HealthSurveyResponse submitHealthSurvey(Long userId, HealthSurveyCreateRequest surveyRequest) {
+        if (healthSurveyRepository.existsByUserId(userId)) {
+            throw new HealthSurveyException(ExceptionType.HEALTH_SURVEY_ALREADY_EXISTS);
+        }
+
+        User user = userService.getUserById(userId);
+        HealthSurvey healthSurvey = mapToEntity(surveyRequest, user);
+
+        healthSurveyRepository.save(healthSurvey);
+        log.info("Saved health survey: {}", healthSurvey.getId());
+
+        return mapToHealthSurveyResponse(healthSurvey);
+    }
+
+    @Transactional
+    public HealthSurveyResponse updateHealthSurvey(Long userId, HealthSurveyUpdateRequest surveyRequest) {
+        HealthSurvey healthSurvey = getExistingHealthSurvey(userId);
+
+        if (surveyRequest.weight() != null) {
+            healthSurvey.updateWeight(surveyRequest.weight());
+        }
+
+        if (surveyRequest.injuries() != null) {
+            healthSurvey.updateInjuries(surveyRequest.injuries());
+        }
+
+        healthSurveyRepository.save(healthSurvey);
+        log.info("Updated health survey for ID: {}", healthSurvey.getId());
+
+        return mapToHealthSurveyResponse(healthSurvey);
+    }
+
+    @Transactional(readOnly = true)
+    public HealthSurveyResponse getHealthSurveyByUserIdResponse(Long userId) {
+        HealthSurvey healthSurvey = getExistingHealthSurvey(userId);
+        log.info("Found health survey: {}", healthSurvey.getId());
+
+        return mapToHealthSurveyResponse(healthSurvey);
+    }
+
+    @Transactional(readOnly = true)
+    public List<String> getAllInjuriesByUserId(Long userId) {
+        List<String> injuries = healthSurveyRepository.findAllInjuriesByUserId(userId).orElseThrow(
+                () -> new HealthSurveyException(ExceptionType.HEALTH_SURVEY_NOT_FOUND));
+        log.info("Found {} injuries.", injuries.size());
+
+        return injuries;
+    }
+
+    @Transactional
+    public void deleteHealthSurveyByUserId(Long userId) {
+        if (!healthSurveyRepository.existsByUserId(userId)) {
```

Comment:
tu ogolna uwaga do wszystkich miejsc tego typu bo widze ze masz tendencje do takich rzeczy. 
Moze oprzyj metody na optionalach i wtedy mozna nie pisac ifologi w kolko takiej samej w kazdje prawie metodzie. tez wszedzie sprawdzasz jakies not nulle. przekmin temat bo nie po to jest optional. + te rzeczy co wypisalem na poczatku zdania